File: 프로그래머스/LV2/코딩테스트 연습/연습문제/호텔 대실/호텔 대실.py
def to_minutes(hhmm):
    hh, mm = hhmm.split(':')
    return int(hh) * 60 + int(mm)


def solution(book_time):
    for i in range(len(book_time)):
        book_time[i][0] = to_minutes(book_time[i][0])
        book_time[i][1] = to_minutes(book_time[i][1])

    book_time.sort()
    rooms = [book_time[0]]

    for i in range(1, len(book_time)):
        for j in range(len(rooms)):
            if rooms[j][1] + 10 <= book_time[i][0]:
                rooms[j][1] = book_time[i][1]
                break
        else:
            rooms.append(book_time[i])
                
    return len(rooms)

File: 프로그래머스/LV2/코딩테스트 연습/연습문제/호텔 대실/호텔 대실.cpp
#include <string>
#include <vector>
#include <sstream>
#include <algorithm>

using namespace std;

vector<string> split(string str, char d) {
    istringstream iss(str);
    string buffer;

    vector<string> result;

    while (getline(iss, buffer, d))
        result.push_back(buffer);

    return result;
}

int toMinutes(string hhmm) {
    vector<string> hm = split(hhmm, ':');
    return stoi(hm[0]) * 60 + stoi(hm[1]);
}

int solution(vector<vector<string>> bookTime) {
    vector<pair<int, int>> bookTimeInt;
    for(int i = 0; i < bookTime.size(); i++)
        bookTimeInt.push_back(make_pair(toMinutes(bookTime[i][0]), toMinutes(bookTime[i][1])));
    

    sort(bookTimeInt.begin(), bookTimeInt.end());
    vector<pair<int, int>> rooms(1, bookTimeInt[0]);
    bool flag;
    
    for(int i = 1; i < bookTimeInt.size(); i++) {
        flag = false;
        for(int j = 0; j < rooms.size(); j++) {
            if(rooms[j].second + 10 <= bookTimeInt[i].first) {
                rooms[j].second = bookTimeInt[i].second;
                flag = true;
                break;
            }
        }
        if(!flag) rooms.push_back(bookTimeInt[i]);
    }

    return rooms.size();
}